### PR TITLE
Implement mdc web layout grid

### PR DESCRIPTION
**Pull request by mthouve**
Fix for this issue: [Implement MDC-Web Layout Grid](https://github.com/secondstreet/ember-material-components-web/issues/7)

This adds Ember-ized component support for the Material Components Web Layout Grids object: https://material.io/components/web/catalog/layout-grids/.

The following examples are added to the [demo page](https://github.com/secondstreet/ember-material-components-web/blob/master/tests/dummy/app/templates/application.hbs).

```hbs
{{#freestyle-section name="Layout Grids" as |section|}}
  {{#freestyle-usage "lg-column-spans" title="Column spans"}}
    {{#mdc-layout-grid as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell span=6}}
          Span 6
        {{/inner.cell}}
        {{#inner.cell span=4}}
          Span 4
        {{/inner.cell}}
        {{#inner.cell span=2}}
          Span 2
        {{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
  {{#freestyle-usage "lg-column-spans-specific" title="Column spans for specific screen sizes"}}
    {{#mdc-layout-grid as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell span=6 span-tablet=8}}
          6 (8 tablet)
        {{/inner.cell}}
        {{#inner.cell span=4 span-tablet=6}}
          4 (6 tablet)
        {{/inner.cell}}
        {{#inner.cell span=2 span-phone=4}}
          2 (4 phone)
        {{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
  {{#freestyle-usage "lg-cell-reordering" title="Cell reordering"}}
    {{#mdc-layout-grid as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell order=3}}
          Cell 3
        {{/inner.cell}}
        {{#inner.cell order=1}}
          Cell 1
        {{/inner.cell}}
        {{#inner.cell order=2}}
          Cell 2
        {{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
  {{#freestyle-usage "lg-cell-alignment" title="Cell alignment"}}
    {{#mdc-layout-grid as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell align="top"}}
          Top aligned
        {{/inner.cell}}
        {{#inner.cell align="middle"}}
          Middle aligned
        {{/inner.cell}}
        {{#inner.cell align="bottom"}}
          Bottom aligned
        {{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
  {{#freestyle-usage "lg-fixed-width-alignment" title="Fixed column width grid, alignment of grid"}}
    {{#mdc-layout-grid fixed-column-width=true align="left" as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell}}
          ...
        {{/inner.cell}}
        {{#inner.cell}}
          ...
        {{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
  {{#freestyle-usage "lg-nested-grid" title="Nested grid"}}
    {{#mdc-layout-grid as |grid|}}
      {{#grid.inner as |inner|}}
        {{#inner.cell}}
          {{#grid.inner as |inner2|}}
            {{#inner2.cell}}Second Level{{/inner2.cell}}
            {{#inner2.cell}}Second Level{{/inner2.cell}}
          {{/grid.inner}}
        {{/inner.cell}}
        {{#inner.cell}}First Level{{/inner.cell}}
        {{#inner.cell}}First Level{{/inner.cell}}
      {{/grid.inner}}
    {{/mdc-layout-grid}}
  {{/freestyle-usage}}
{{/freestyle-section}}
```

### CODE REVIEW

**Comment by Kerrick** [comment]
Nested grids should work too, right?

```hbs
{{#mdc-layout-grid as |grid|}}
  {{#grid.inner as |inner|}}
    {{#inner.cell}}
      {{#grid.inner}}
        {{#grid.cell}}Second Level{{/grid.cell}}
        {{#grid.cell}}Second Level{{/grid.cell}}
      {{/grid.inner}}
    {{/inner.cell}}
    {{#grid.cell}}First Level{{/grid.cell}}
    {{#grid.cell}}First Level{{/grid.cell}}
  {{/grid.inner}}
{{/mdc-layout-grid}}
```

File: addon/components/mdc-layout-grid.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    'isAlignRight:mdc-layout-grid--align-right'
+  ],
+  layout,
+  //endregion
+
+  //region Attributes
+  /**
+   * @type {Boolean}
+   */
+  'fixed-column-width': false,
```
Comment (Kerrick) [inline addon/components/mdc-layout-grid.js]: If possible can you please list the `Attributes` before the `Ember Hooks`? It helps people who dive into the code to find the "configurable" parts more easily.

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "@material/fab": "0.3.16",
     "@material/form-field": "0.2.14",
     "@material/icon-toggle": "0.1.19",
+    "@material/layout-grid": "^0.4.4",
```
Comment (Kerrick) [inline package.json]: In order to avoid breakage, I recommend locking to a specific version rather than allowing npm to upgrade via `^`.

Comment (mthouve) [reply package.json]: Done!

File: addon/components/mdc-layout-grid.js (hunk 1 above)
Comment (mthouve) [reply addon/components/mdc-layout-grid.js]: Done!

**Comment by Kerrick** [comment]
Thank you so much for the PR! This will ship out with v0.0.37 👍

**Pull request merged**